workers/unity/Assets/Gdk/Core/Collections/MapProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    internal static class MapProvider<TKey, TValue>
```

Comment:
`Map` is a C++/schema name - would `Dictionary` be more in-line with C# convention?